sdk/servicebus/service-bus/samples/typescript/src/advanced/sessionManagerSample.ts

```diff
@@ -0,0 +1,155 @@
+import { ServiceBusClient, delay, ReceivedMessageWithLock } from "@azure/service-bus";
+import * as dotenv from "dotenv";
+import { env } from "process";
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+
+dotenv.config();
+
+const serviceBusConnectionString =
+  env["SERVICEBUS_CONNECTION_STRING"] || "<service bus connection string not in environment>";
+const queueName = env["QUEUE_NAME_WITH_SESSIONS"] || "<queue name not in environment>";
+const maxSessionsToProcessSimultaneously = 8;
+const sessionIdleTimeoutMs = 3 * 1000;
+const delayWhenNoSessionsAvailableMs = 5 * 1000;
+
+interface SessionMessageHandlers {
+  processInitialize(sessionId: string);
+  processMessage(msg: ReceivedMessageWithLock);
+  processError(err: Error, sessionId: string);
+  processClose(reason: "error" | "idle_timeout", sessionId: string);
+}
+
+function _createIdleTimer(
```

Comment:
It would be nice if this was supported out of the box by the session receiver. But I can see that the challenge is going to be communicating to the caller that we stopped receiving messages because of the timeout vs any kind of error.

Reply:
Maybe we could just export it as a primitve, similar to what we do for `delay`?

Reply:
I made it a bit shorter which should hopefully ease the pain :)